### PR TITLE
Add `toml` LSP

### PR DESCRIPTION
**Pull request by kariy**
Release Notes:

- Added `toml` LSP using [taplo](https://taplo.tamasfe.dev/)

### CODE REVIEW

File: crates/zed/src/languages/toml.rs [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+async fn get_cached_server_binary(container_dir: PathBuf) -> Option<LanguageServerBinary> {
+    async_maybe!({
+        let mut last = None;
+        let mut entries = fs::read_dir(&container_dir).await?;
+        while let Some(entry) = entries.next().await {
+            last = Some(entry?.path());
+        }
+
+        anyhow::Ok(LanguageServerBinary {
+            path: last.ok_or_else(|| anyhow!("no cached binary"))?,
```
Comment (SomeoneToIgnore) [inline crates/zed/src/languages/toml.rs:108]: There's two new `ok_or_else(|| anyhow!(` calls in this PR, and both could be just `foo.context("bar")?`, let's simplify both?

**Review by SomeoneToIgnore — COMMENTED** [review]
Looks great!
One thing that bugs me is that after formatting, the caret is being moved to the new line which no other formatters did to me before, but not sure how to fix it here.

Let's reflect the changes in the docs: https://github.com/zed-industries/zed/blob/main/docs/src/languages/toml.md and fix the style nit, and we're good to merge this.

File: crates/zed/src/languages/toml.rs (hunk 1 above)
Comment (kariy) [reply crates/zed/src/languages/toml.rs]: I just copied it from `rust.rs`, but I've updated it now.

**Review by SomeoneToIgnore — APPROVED** [review]
Thank you!

**Pull request merged**